### PR TITLE
Add zio-test to TestModule

### PR DESCRIPTION
**Pull request by carlosedp**
This PR adds ZIOTest framework to TestModule.

### CODE REVIEW

File: readme.adoc [hunk 1]
```diff
@@ -932,7 +932,7 @@ _Changes since {prev-version}:_
 * `ScalaModule`: Support for Scala 3
 * `CoursierModule`: Support customized dependency resolution (needed to work with ScalaFX)
 * `TestModule`: Added new `testFramework` target and only support one test framework. Deprecated `testFrameworks` targets.
-* `TestModule`: Added new convenience traits to configure popular test frameworks, e.g. `TestModule.Junit4`, `TestModule.ScalaTest`, `TestModule.Utest`, and many more
+* `TestModule`: Added new convenience traits to configure popular test frameworks, e.g. `TestModule.Junit4`, `TestModule.ScalaTest`, `TestModule.Utest`, `TestModule.ZIOTest` and many more
```
Comment (lefou) [inline readme.adoc:935]: Wrong place. This is the changelog of version 0.9.7.

**Review by lefou — COMMENTED** [review]
Nice addition, thanks.

File: readme.adoc (hunk 1 above)
Comment (carlosedp) [reply readme.adoc]: My bad, removed on last commit.

**Review by lefou — COMMENTED** [review]
I'd prefer `ZioTest` instead of `ZIOTest`, which is more in line with other naming in Mill (except `ScalaJS` for compatibility reasons), e.g. `TestNg` vs. `TestNG`. But I'm open for arguments.

**Comment by carlosedp** [comment]
I agree, this makes trait naming more homogeneous.

File: scalalib/src/mill/scalalib/TestModule.scala [hunk 2]
```diff
@@ -242,6 +242,14 @@ object TestModule {
     override def testFramework: T[String] = "weaver.framework.CatsEffect"
   }
 
+  /**
+   * TestModule that uses ZIO Test Framework to run tests.
+   * You need to provide the zio-test dependencies yourself.
+   */
+  trait ZioTest extends TestModule
```
Comment (lefou) [inline scalalib/src/mill/scalalib/TestModule.scala:249]:
```suggestion
  trait ZioTest extends TestModule {
```

**Pull request merged**